From f9cc375ed6dfcd51513d99a21175e4ac669f30c8 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Tue, 18 Mar 2025 18:23:36 +0100
Subject: [PATCH] arm64: dts: fix csi data-lanes property in stm32mp257f-dk.dts

The csi receiver data-lanes property should always start from 1
instead of 0 since 0 is reserved for the clock-lane. Update the
device-tree to match this requirement.

Change-Id: I7c0f58b7ad3356688c64487c6c339e03985e6ccc
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/445122
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
Tested-by: Alain VOLMAT <alain.volmat@st.com>
Reviewed-by: Alain VOLMAT <alain.volmat@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index cb40de9ff97e..cff4e8c8c26f 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -236,7 +236,7 @@ port@0 {
 			reg = <0>;
 			csi_sink: endpoint {
 				remote-endpoint = <&imx335_ep>;
-				data-lanes = <0 1>;
+				data-lanes = <1 2>;
 				bus-type = <4>;
 			};
 		};
-- 
2.50.1

